### PR TITLE
Amos : correction des bug

### PR DESCRIPTION
**Pull request by amosjerome**
6 bugs detected and corrected

### CODE REVIEW

File: stack.hpp [hunk 1]
```diff
@@ -16,19 +16,65 @@ class Stack {
     }
 
     void push(const T& element) {
-        if (position_ > capacity_) throw std::out_of_range("Not enough capacity");
-        storage_[position_++] = element;
+     /*   if (position_ < capacity_) // -1 add for the check *************
```
Comment (cm0x4D) [inline stack.hpp]: There is no need to keep the old code in comments, if we ever want to go back, GIT has everything we need...

**Comment by cm0x4D** [comment]
Thanks. Still one (bonus) bug is left. Hint: It is a memory leak.

You did not fix every bug in a single commit. It is good practice to keep commits as atomic as possible.

**Comment by amosjerome** [comment]
fixed the memory bug with a destructor
delete all the old code

File: stack.hpp [hunk 2]
```diff
@@ -74,8 +58,16 @@ class Stack {
...
             return std::move(storage_[0]);
         }
-}
+        }
 
+    ~Stack()
+    {
+        if (storage_ != NULL)
+        {
+            delete storage_;
```
Comment (cm0x4D) [inline stack.hpp]: Wrong delete, it is an array after all.

FYI: If you call delete with a NULL pointer it does just nothing, so the if is not required.

**Pull request closed without merging**